test/utils/string_extension_test.dart

```diff
@@ -377,4 +377,65 @@ void main() {
       expect('<a href="https://example.com"></a>'.extractInnerText(), isEmpty);
     });
   });
+
+  group('[isRoomAlias TEST]', () {
+    final testMap = <String, bool>{
+      '#room:server.com': true,
+      '#room:192.168.1.1': true,
+      '#room:[2001:db8:85a3:8d3:1319:8a2e:370:7348]': true,
+      '#room:example.com:8080': true,
+      '!room:server.com': false,
+      'room:server.com': false,
+      '': false,
+    };
+
+    for (final entry in testMap.entries) {
+      test('Testing: ${entry.key} => Expected: ${entry.value}', () {
+        final result = entry.key.isRoomAlias();
+        if (entry.value) {
+          expect(result, isTrue);
+        } else {
+          expect(result, isFalse);
+        }
+      });
+    }
+  });
+
+  group('[getServerNameFromRoomAlias] TEST', () {
+    final testMap = <String, String?>{
+      '#room:server.com': 'server.com',
+      '#room:domain.com:': 'domain.com',
+      '': null,
+    };
+
+    for (final entry in testMap.entries) {
+      test('Testing: ${entry.key} => Expected: ${entry.value}', () {
+        final result = entry.key.getServerNameFromRoomIdOrAlias();
+        expect(result, equals(entry.value));
+      });
+    }
+  });
+
+  group('[isRoomId TEST]', () {
+    final testMap = <String, bool>{
+      '!room:server.com': true,
+      '!room:192.168.1.1': true,
+      '!room:[2001:db8:85a3:8d3:1319:8a2e:370:7348]': true,
+      '!room:example.com:8080': true,
+      '#room:server.com': false,
+      'room:server.com': false,
+      '': false,
+    };
+
+    for (final entry in testMap.entries) {
+      test('Testing: ${entry.key} => Expected: ${entry.value}', () {
+        final result = entry.key.isRoomId();
+        if (entry.value) {
```

Comment:
IMO you can replace that
```dart
if (entry.value) {
    expect(result, isTrue);
} else {
    expect(result, isFalse);
}
```
with this
```dart
expect(result, entry.value);
```
something like that

Reply:
updated